### PR TITLE
Feature/faction model refactoring

### PR DESCRIPTION
**Pull request by ssipasseuth**
Full refactoring + bug fixes.
This also include modification to allow "smooth" refreshing in the item creator

### CODE REVIEW

File: views/js/pciCreator/dev/fractionModelInteraction/pciCreator.json [hunk 1]
```diff
@@ -19,5 +19,6 @@
     "response": {
         "baseType": "integer",
         "cardinality": "single"
-    }
-}
+    },
+    "debug" : true
```
Comment (clawfire) [inline views/js/pciCreator/dev/fractionModelInteraction/pciCreator.json]: You should not commit that

File: views/js/pciCreator/dev/fractionModelInteraction/runtime/fractionModelInteraction.js [hunk 2]
```diff
@@ -2,12 +2,36 @@ define([
...
+
+        var canvasSize = 2 * (parseInt(config.radius) + parseInt(config.padding) + parseInt(config.outlineThickness));
+        var canvas = scaleRaphael($('.shape-container', $container)[0], canvasSize, canvasSize);
+
+        //make it resizeable:
+//        var containerWidth = $container.innerWidth();
+//        var height = canvasSize;
+//        var factor = 1;
+//        canvas.changeSize(containerWidth, height * factor, false, false);
+//        canvas.scaleAll(factor);
```
Comment (clawfire) [inline views/js/pciCreator/dev/fractionModelInteraction/runtime/fractionModelInteraction.js]: if it's commented, you can delete it. It's versionned ;)

File: views/js/pciCreator/dev/fractionModelInteraction/runtime/fractionModelInteraction.js [hunk 3]
```diff
@@ -20,74 +44,150 @@ define([
...
-                event.data.numerator = event.data.config.selectedPartitionsInit;
-                event.data.denominator = event.data.config.partitionInit;
-                $(event.data.dom).trigger('reset.fraction');
-            }).on('changeResponse.fraction reset.fraction',this,function(event){
+
+            }).on('click', 'button.reset', function(){
+
+                _this.setFractionModel(_this.config.selectedPartitionsInit, _this.config.partitionInit);
+                _this.setState(_this.config.selectedPartitions);
+                $container.trigger('reset.fraction');
```
Comment (clawfire) [inline views/js/pciCreator/dev/fractionModelInteraction/runtime/fractionModelInteraction.js:153]: That's ok for me but I think we should namespace events

**Comment by ssipasseuth** [comment]
Your comments make sense so I apply your suggestion on the latest commit.

**Pull request merged**